packages/light-client/README.md

```diff
@@ -88,6 +91,31 @@ lightclient.emitter.on(LightclientEvent.lightClientOptimisticHeader, async (opti
 });
 ```
 
+## Browser Integration
+
+If you want to use Lightclient in browser and facing some issues in building it with bundlers like webpack, vite. We suggest to use our distribution build. The support for single distribution build is started from `1.18.0` version.
+
+Directly link the dist build with the `<script />` tag with tools like unpkg or other. e.g.
+
+```html
+<script src="https://www.unpkg.com/@lodestar/light-client@1.18.0/dist/lightclient.es.min.js" type="module">
+```
+
+Then the lightclient package will be exposed to `globalThis`, incase of browser environment that will be `window`. You can access the package as `window.lodestar.lightclient`. All named exports will also be available from this interface. e.g. `window.window.lodestar.lightclient.transport`.
+
+NOTE: Due to `top-level-await` used in one of dependent library, the package will not be available right after the load. You have to use a hack to clear up that await from the event loop.
+
+```html
```

Comment:
Would be nice to find a way to remove the need for this

Reply:
It's in the `bls` library. @matthewkeil is working on a newer version. Hope that will solve this problem.

Reply:
it will not. there will always be a top level await in esm because the bindings path is programmatic so its not imported it looked up and then `await import(bindingLocation)`

Reply:
i suppose it would be possible in `bls` to use hard imports and then `null-loader` from weback to not include the other version for web.  But for blst if we make it esm it will always have the top-level await for the import

Reply:
It's possible avoid top-level-import, but then would have to leave upto user of the library to init where appropriate.

Reply:
> it will not. there will always be a top level await in esm because the bindings path is programmatic so its not imported it looked up and then await import(bindingLocation)

I think there is a difference between using top-level await and a ESM module without it. Eg. Node 22 will ship a new feature to `require` ES modules with the exception of the ones that use top-level await (see https://github.com/nodejs/node/pull/51977).

In any case, we might wanna avoid top-level await in packages that are used by others.

Reply:
I made the top-level await go away in `blst` fyi.  so might be able to also do something similar in `bls`...

https://github.com/ChainSafe/blst-ts/blob/37b13881307ee4eb8fb8a4a84f6cafccfde5293c/lib/index.mjs#L7-10